src/compile.js

```diff
@@ -1259,15 +1266,136 @@ Compiler.prototype.ctryexcept = function (s) {
 
     this.setBlock(orelse);
     this.vseqstmt(s.orelse);
-    // Should jump to finally, but finally is not implemented yet
     this._jump(end);
+
     this.setBlock(end);
 };
 
+Compiler.prototype.outputFinallyCascade = function (thisFinally) {
+    var nextFinally;
+    
+    if (this.u.finallyBlocks.length == 0) {
+        out("if($postfinally!==undefined) { if ($postfinally.returning) { return $postfinally.returning; } else { $blk=$postfinally.gotoBlock; $postfinally=undefined; continue; } }");
+    } else {
+        // If this was a nonlocal return, we need to work out where to go.
+        // If it's a return, we always continue the finally chain.
+        // If it's a break or continue, we need to work out whether our parent
+        // finally-block is at the same break/continue depth as us (ie whether there's
+        // a loop/breakable-block boundary between us and it).
+
+        nextFinally = this.peekFinallyBlock();
+        
+        out("if($postfinally!==undefined) {",
```

Comment:
(Updated to match the fact that `break` and `continue` depths are always the same: I now just track `breakDepth`, and I no longer distinguish between `isBreak` and `isContinue` in `$postfinally`.)